src/libs/Violations/ViolationsUtils.ts

```diff
@@ -64,6 +65,17 @@ function getTagViolationsForMultiLevelTags(
         (violation) => violation.name !== CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED && violation.name !== CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
     );
 
+    if (hasDependentTags && !updatedTransaction.tag) {
```

Comment:
Since it isn't super obvious _why_ we do this, and given the description in the js doc for this method is now outdated, I think we should split the logic here into getTagViolationsForDependentTags and getTagViolationsForIndependentTags

Then the js doc can stay as is, and the one for getTagViolationsForDependentTags can say something similar to the comment I suggested in getViolationsForField

/**
 * Calculates missing tag violations for policies with dependent tags, 
 * by returning one per tag with its corresponding tagName in the data
 */